WISE-Community/WISE
# JSPM to WebPack

This commit removes JSPM and *.es6 and *.js.map files in WISE5 applications (vle, cm, at). This will also change the way we develop WISE5 applications. Instead of having gulp watch and transpile changes, it uses the webpack dev server. Each application will run on its own server (vle: 8081, cm: 8082, at: 8083)

How to test:
1. Remove src/main/webapp/wise5/jspm_modules folder
1. Run npm install
1. Start WISE as before (localhost:8080)
1. Start builds for vle/cm/at/site ```npm run build-dev-all```
1. Make sure that you can develop and test as before for all applications (vle/cm/at/site), including
   - Saving/loading student data, annotations, curriculum
   - WebSockets
1. Shutdown npm run build-dev-all process (ctrl-c)
1. Start VLE on port 8081 ```npm run standalone-server-vle```
1. Start CM on port 8082 ```npm run standalone-server-cm```
1. Start AT on port 8083 ```npm run standalone-server-at```
1. Preview a project by going to localhost:8081/#/project/456
1. In browser 1, log in as student for run id = 123, for example
1. In browser 1, go to localhost:8081/#/run/123. Student VLE should load.
1. In browser 2 (incognito or another browser), log in as teacher for run id = 123.
1. In browser 2, go to localhost:8082. When prompted for run id, type in 123. Classroom Monitor should load.
1. In browser 2, go to localhost:8083. Authoring Tool should load.
1. Make sure that things work as before, including
   - Saving/loading student data, annotations, curriculum
   - WebSockets
1. Make sure that making changes to WISE5 js files automatically builds and reloads the application
1. Kill VLE, CM, and AT processes.
1. Run ```npm run build-all```. This should build vle, cm, at, and site and put them in the respective applications' ```dist``` folder.
1. Test localhost:8080 as if you are running WISE on production.

Closes #2004
